### PR TITLE
[py-watchdog] added version 2.1.6

### PR DESCRIPTION
**Pull request by qwertos**

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-watchdog/package.py [hunk 1]
```diff
@@ -10,19 +10,16 @@ class PyWatchdog(PythonPackage):
...
-    version('0.10.3', sha256='f7e5040b483cc9a8618a4e72c11a527b42434a0766ea96dce633e8b44b1369de')
-    version('0.10.2', sha256='4dba861f5e6960c8063ad68137772ff35f1516ea47d64b53042dabd2d8f8dbdc')
-    version('0.10.1', sha256='972c6050eb5d275b582c75e6ff71ef562a6c3be6361d6d9b37110e0ba718994d')
-    version('0.10.0', sha256='39e2966b8c9596e45b463815668989b87c7d205e47c7e6e4a7db9a58354b99ff')
-    version('0.9.0',  sha256='e8a32701dff43a4e671a40acf92f55a8fffd8ca74333b75436d63a94d104faef')
-    version('0.8.3',  sha256='e9a27d0ab121fc86217ab833e778c76880aad4ecdb623796f4b7aee9925394ed')
-    version('0.8.2',  sha256='386e882c8bc7df8a3c4c8803e6fcf2f7cf836c9a8867ff9d91b4af19d262d023')
-    version('0.8.1',  sha256='05e5d8acd5061aff789359cd8f25a85ba4c848a8feded3fc68f9c57f0b181373')
-    version('0.8.0',  sha256='5abac06e63ad8d5b001626a16bfdd1b918637aa8d85cf933e68de2e627b56053')
-    version('0.7.1',  sha256='d795fa85ce9252eeb2294a5182c99013433aeb736cc7a1fc9e14e1e2a1a19690')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-watchdog/package.py:24]: Can you keep these versions and either update the checksums or mark them as deprecated? We don't want to remove them without deprecating them first. Spack's download mirror still has all these versions, so people can still use them.

**Review by adamjstewart — CHANGES_REQUESTED** [review]

File: var/spack/repos/builtin/packages/py-watchdog/package.py (hunk 1 above)
Comment (qwertos) [reply var/spack/repos/builtin/packages/py-watchdog/package.py:24]: Do the dependencies have to be accurate for the versions marked as deprecated?

Comment (adamjstewart) [reply var/spack/repos/builtin/packages/py-watchdog/package.py:24]: Preferably... I would be fine with more strict dependencies though. For example, if older deprecated versions require foo@1.2+ and newer non-deprecated versions require foo@1.3+ I would be fine with saying all versions require foo@1.3+. That way, all versions still build correctly, although the version requirements are more strict than necessary for older versions.

File: var/spack/repos/builtin/packages/py-watchdog/package.py [hunk 2]
```diff
@@ -10,19 +10,49 @@ class PyWatchdog(PythonPackage):
...
+    version('0.8.3',  sha256='7e65882adb7746039b6f3876ee174952f8eaaa34491ba34333ddf1fe35de4162')
+    version('0.8.2',  sha256='33a9ab3ce2e6b1aca4d2a50752231668d69bdba4ab096d9742195ccfbef1e023')
+    version('0.8.1',  sha256='d6ec6be582b244834a888c8ccc2d451816184ab104b5454b5e5cd7649e8f671c')
+    version('0.8.0',  sha256='a86bb2d8b94bb4bf76fcc2ff36f741c0e511ec24c4d3a1059b47d49e377d64f5')
+    version('0.7.1',  sha256='54ca64fdf0a2fb23cecba6349f9587e62fd31840ae22a71898a65adb8c6b52f9')
+
+    variant('watchmedo', default=False, description="Build optional watchmedo utility script")
+
+    # Package assumes you want +watchmedo when @:0.9.0.
+    conflicts('~watchmedo',      when='@:0.9.0')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-watchdog/package.py]:
```suggestion
    variant('watchmedo', default=False, when='@0.10:', description="Build optional watchmedo utility script")
```

**Review by adamjstewart — CHANGES_REQUESTED** [review]

**Review by adamjstewart — DISMISSED** [review]

**Comment by adamjstewart** [comment]
flake8:
```
  var/spack/repos/builtin/packages/py-watchdog/package.py:50: [E501] line too long (91 > 88 characters)
  var/spack/repos/builtin/packages/py-watchdog/package.py:51: [E501] line too long (94 > 88 characters)
```

**Review by adamjstewart — APPROVED** [review]

**Pull request merged**